ts/src/header-validator/validate.ts

```diff
@@ -168,26 +168,26 @@ export enum ItemErrorAction {
   earlyExit,
 }
 
-function isCollection<P extends PathComponent, V, C extends Context>(
+function makeCollection<Col, P extends PathComponent, V, C extends Context>(
```

Comment:
```suggestion
function collection<Col, P extends PathComponent, V, C extends Context>(
```

I'm wondering if just `collection` would be better aligned with the other naming.